### ISSUE TITLE
[DEMO 3] Fixes and improvements before demo

### ISSUE THREAD

**Issue by ciaccogiuseppe** [GigiLoria5/se2022-08-HireTracker#78]
After story8 will be merged into dev, there are still some things to be fixed before releasing the next version of the app

- [x] Fixing reference points in existing hikes: they have to be points **on** the track and not outside it
- [x] Adding hikes to the database (as many as possible)
- [x] Removing warnings from node console
- [ ] Phone number are not validated correctly
- [x] Description field should not be present for hut and parking_lot
- [x] Missing end point marker on reference point add map

other tasks will be added if needed

**Comment by GigiLoria5** [comment]
I would also add removing warnings from the node console... are annoying (at least to see)
![immagine](https://user-images.githubusercontent.com/41837640/206893072-a2d126de-be2a-474a-bf65-25b2e0adec36.png)

**Comment by GigiLoria5** [comment]
The add hut has still some problems in validating the fields... I suppose that some checks is not done correctly. EDIT: only mobile phone are valid... "home phone" are considered invalid in the backend...
![immagine](https://user-images.githubusercontent.com/41837640/206904177-0168b997-eb66-4803-8ed5-3be75d00250f.png)

**Comment by GigiLoria5** [comment]
The map where is possible to add reference points does not have the "end point marker"
![immagine](https://user-images.githubusercontent.com/41837640/206904205-97c8d680-d7b6-4eed-8b72-6d2f2daefc1f.png)

**Comment by GigiLoria5** [comment]
Hut and parking lot in the process of adding a new hike, must not have the description field as this field is not needed for them (all the information about them already exist)
![immagine](https://user-images.githubusercontent.com/41837640/206905012-88105668-b297-40c3-a355-d4394fb8ef38.png)

**Comment by GigiLoria5** [comment]
Hikes added by @michelepistan

**Comment by GigiLoria5** [comment]
Description fields removed by @michelepistan

**Comment by GigiLoria5** [comment]
'Missing end point marker on reference point add map' fixed

**Comment by GigiLoria5** [comment]
The phone number will be handled better in the next sprint
